stdlib/public/core/RangeReplaceableCollectionType.swift

```diff
@@ -446,10 +446,10 @@ public func +<
     S : SequenceType
     where S.Generator.Element == C.Generator.Element
 >(lhs: C, rhs: S) -> C {
-  var lhs = lhs
-  // FIXME: what if lhs is a reference type?  This will mutate it.
-  lhs.appendContentsOf(rhs)
-  return lhs
+  var result = lhs is AnyObject ? C(lhs) : lhs
```

Comment:
We could just use `var result = C(lhs)`, as values type collections will generally be COW anyways so it would be equivalent to the plain assignment.